src/Compilers/Core/Portable/Text/CompositeText.cs

```diff
@@ -77,7 +75,8 @@ internal override ImmutableArray<SourceText> Segments
 
         public override SourceText GetSubText(TextSpan span)
         {
-            CheckSubSpan(span);
+            if (!ValidateSubSpan(span))
+                return From(string.Empty, Encoding, ChecksumAlgorithm);
```

Comment:
i don't get this logic.  if we can't validate, we return nothing?  why not just validate, then check if hte span length is 0 *in* this method?

in other words, the above pattern is pretty wonky to me.

Reply:
I used this pattern because that's how the existing `CheckCopyToArguments` method works. But I agree that it's a little odd. I can remove the length check.

Reply:
Fixed.